Gitweb:     http://git.kernel.org/linus/2eb2fa67e5462a36e98172fb92c78bc405b3035f
Commit:     2eb2fa67e5462a36e98172fb92c78bc405b3035f
Parent:     634a555ce3ee5ea1fdcaee8b4ac9ce7b54f301ac
Author:     Bob Copeland <me@bobcopeland.com>
AuthorDate: Mon Nov 16 08:30:29 2009 -0500
Committer:  John W. Linville <linville@tuxdriver.com>
CommitDate: Mon Nov 16 14:17:14 2009 -0500

    ath5k: allow setting txpower to 0
    
    As a holdover from earlier code when we used to set
    the power limit to '0' after a reset to configure the
    default transmit power, ath5k interprets txpower=0 as
    12.5 dBm.  Fix that by just passing 0 through.
    
    This fixes http://bugzilla.kernel.org/show_bug.cgi?id=14567
    
    Cc: stable@kernel.org
    Reported-by: Daniel Folkers <daniel.folkers@task24.nl>
    Tested-by: Daniel Folkers <daniel.folkers@task24.nl>
    Signed-off-by: Bob Copeland <me@bobcopeland.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>
---
 drivers/net/wireless/ath/ath5k/phy.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/ath/ath5k/phy.c
+++ b/drivers/net/wireless/ath/ath5k/phy.c
@@ -2954,8 +2954,6 @@ ath5k_hw_txpower(struct ath5k_hw *ah, st
 		ATH5K_ERR(ah->ah_sc, "invalid tx power: %u\n", txpower);
 		return -EINVAL;
 	}
-	if (txpower == 0)
-		txpower = AR5K_TUNE_DEFAULT_TXPOWER;
 
 	/* Reset TX power values */
 	memset(&ah->ah_txpower, 0, sizeof(ah->ah_txpower));
